### PR TITLE
code suggestions

### PR DESCRIPTION
**Pull request by leafty**
@rokroskar I think you can just merge this into #77

### CODE REVIEW

File: components/renku_data_services/utils/core.py [hunk 1]
```diff
@@ -12,3 +15,22 @@ def get_ssl_context():
...
+        type_conflict_strategies=["override_if_not_empty"],
+    )
+
+    merged_spec: dict[str, Any]
+    merged_spec = functools.reduce(merger.merge, args, dict())
+
+    # Remove duplicate entries in `.servers`
+    server_urls: Set[str]
+    server_urls = set((obj["url"] for obj in merged_spec.get("servers", [])))
+    merged_spec["servers"] = [dict(url=url) for url in server_urls]
```
Comment (rokroskar) [inline components/renku_data_services/utils/core.py:34]: I don't think you need this - check https://renku-ci-ds-77.dev.renku.ch/swagger/?urls.primaryName=data%20service

image

File: components/renku_data_services/utils/core.py [hunk 2]
```diff
@@ -12,3 +15,22 @@ def get_ssl_context():
     if custom_cert_file:
         context.load_verify_locations(cafile=custom_cert_file)
     return context
+
+
+def merge_api_specs(*args):
+    """Merges API spec files into a single one."""
+    merger = Merger(
+        type_strategies=[(list, "append_unique"), (dict, "merge"), (set, "union")],
```
Comment (rokroskar) [inline components/renku_data_services/utils/core.py:23]: is `append_unique` a built-in strategy?

**Review by rokroskar — COMMENTED** [review]
Thanks @leafty ! Just two quick questions

File: components/renku_data_services/utils/core.py (hunk 2 above)
Comment (leafty) [reply components/renku_data_services/utils/core.py:23]: Yes -> https://github.com/toumorokoshi/deepmerge/blob/v1.1.0/deepmerge/strategy/list.py

File: components/renku_data_services/utils/core.py (hunk 1 above)
Comment (leafty) [reply components/renku_data_services/utils/core.py]: Done in 2759e23cf9c07ba1bd20ce14df1317bc1fe76934.

File: components/renku_data_services/utils/core.py (hunk 2 above)
Comment (rokroskar) [reply components/renku_data_services/utils/core.py:23]: I guess their docs need an update 😆

**Review by rokroskar — APPROVED** [review]

File: components/renku_data_services/utils/core.py (hunk 2 above)
Comment (leafty) [reply components/renku_data_services/utils/core.py:23]: I didn't know they had docs :rofl: I just Ctrl+Click in vscode.

Comment (rokroskar) [reply components/renku_data_services/utils/core.py:23]: obvs the better approach 😆 🙈

**Pull request merged**